api/src/services/users/create_saved_opportunity.py

```diff
@@ -0,0 +1,25 @@
+from uuid import UUID
+
+from sqlalchemy import select
+
+from src.adapters import db
+from src.db.models.user_models import UserSavedOpportunity
+
+
+def create_saved_opportunity(db_session: db.Session, user_id: UUID, json_data: dict) -> None:
+    # Check if the record exists
+    record = db_session.execute(
+        select(UserSavedOpportunity).filter(
+            UserSavedOpportunity.user_id == user_id,
+            UserSavedOpportunity.opportunity_id == json_data["opportunity_id"],
+        )
+    ).scalar_one_or_none()
+
+    if record:
+        record.is_deleted = False
+    else:
```

Comment:
The if-else here is a good case for something we'd want to be able to easily track in case a user hits a weird issue. I'd recommend adding a log message for each case.